### DISCUSSION TITLE
SlimTreeSequence "can't set attribute" errors

### DISCUSSION THREAD

**Discussion by mshpak76** [tskit-dev/pyslim#269] [Q&A]
I have been trying to use SlimTreeSequence() to converts tree sequence files generated by Slim 3.5 into something that pyslim 0.700 could process. I'm getting a generic "can't set attribute" error, suggesting there's a failed linkage to some other library, but I can't figure out what that dependency may be. In case it matters, this is running on Python 3.8.6

    Traceback (most recent call last):
    File "recapitation_convert_format.py", line 7, in <module>
    ts_new = pyslim.SlimTreeSequence(ts)
    File "/opt/miniconda3/lib/python3.8/site-packages/pyslim/slim_tree_sequence.py", line 203, in __init__
     self.individual_locations = ts.tables.individuals.location
     AttributeError: can't set attribute

**Comment by petrelharp** [comment]
Ah - you'll hit this error if you use the new tskit with pyslim 0.700. I recommend using the beta for pyslim 1.0: https://pypi.org/project/pyslim/1.0b1/

Alternatively, just don't use SlimTreeSequence: pretty much all the functionality provided by SlimTreeSequence has been subsumed into tskit now, so you don't need that class at all. Stuff that isn't is now methods, e.g. `pyslim.recapitate(ts)` instead of `SlimTreeSequence(ts).recapitate()`.

**Reply by jahink17** [reply]
Hi, it seems that the recapitate() method still includes a call to SlimTreeSequence(recap) at the very end and therefore runs into the same problem as above (i.e. the attribute error) -- at least in the version I'm using. Is there any way to work around this? 

Thank you for your help!
